corehq/apps/accounting/forms.py

```diff
@@ -1854,27 +1861,23 @@ def trigger_customer_invoice(self):
 
     @staticmethod
     def clean_previous_invoices(invoice_start, invoice_end, account):
-        invoices = CustomerInvoice.objects.filter(
+        prev_invoices = CustomerInvoice.objects.filter(
             date_start__lte=invoice_end,
             date_end__gte=invoice_start,
             account=account
         )
-        prev_invoices = []
-        for invoice in invoices:
-            if invoice.account == account:
-                prev_invoices.append(invoice)
         if prev_invoices:
             from corehq.apps.accounting.views import CustomerInvoiceSummaryView
             raise InvoiceError(
                 "Invoices exist that were already generated with this same "
                 "criteria. You must manually suppress these invoices: "
                 "{invoice_list}".format(
-                    num_invoices=len(prev_invoices),
+                    num_invoices=prev_invoices.count(),
```

Comment:
If you're doing this for performance, it should be one less SQL query to evaluate the queryset in ```len()```.  Django querysets are cached, so you'll skip running an extra query later if you know you'll eventually need to load the entire queryset.  If you're doing this for some other reason, carry on.

See https://docs.djangoproject.com/en/1.11/ref/models/querysets/#when-querysets-are-evaluated and https://docs.djangoproject.com/en/1.11/ref/models/querysets/#all

Reply:
Oh, cool. Thanks